practice/validator/site.js

```diff
@@ -0,0 +1,118 @@
+'use strict';
+
+// 1. Input
+/**
+ * Gathers what has been typed in the 'name' field for each event.
+ */
+function getUserName() {
+  return $('#name').val();
+}
+/**
+ * Gathers what has been typed in the 'DOB' field for each event.
+ */
+function getUserDOB() {
+  return $('#dob').val();
+}
+/**
+ * Gathers what has been typed in the 'Phone' field for each event.
+ */
+function getUserPhone() {
+  return $('#phone').val();
+}
+
+// 2. Transform Functions
+// No transform.
+
+// 3. Create
+var NAME_MATCH = /^([a-z]+\s[a-z]+)?$/i;
+var DOB_MATCH = /^(\d{4}-\d{2}-\d{2})?$/;
+var PHONE_MATCH = /^(\d{3}-\d{3}-\d{4})?$/;
+/**
+ * Selects the 'name' input box.
+ */
+function getNameBox() {
+  return $('#name');
+}
+/**
+ * Selects the 'dob' input box.
+ */
+function getDOBBox() {
+  return $('#dob');
+}
+/**
+ * Selects the 'phone' input box.
+ */
+function getPhoneBox() {
+  return $('#phone');
+}
+
+// 4. Modify and Sync
+/**
+ * Turns on the invalid class to mark that the input text is invalid.
+ */
+function addInvalid(box) {
+  box.attr('class', 'invalid');
```

Comment:
Since it's possible that there might be other classes applied to this element. `.addClass('invalid')` handles all of the edge cases.